src/components/DataTable/DataTable.tsx

```diff
@@ -97,12 +109,13 @@ export const DataTable = <Data,>({
         error={error}
         loading={loading}
         entityName={entityName}
-        empty={{
-          show: rows.length === 0,
-          textFilter: ensureString(table.getState().globalFilter),
-          hasFilters:
-            data.length !== 0 && table.getState().columnFilters.length > 0,
-        }}
+        empty={
+          isObject(empty) ? { ...defaultEmptyProps, ...empty }
+            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+          : empty === undefined ?
+            defaultEmptyProps
+          : empty
```

Comment:
```suggestion
          : typeof empty === "boolean" ? 
            empty 
            : defaultEmptyProps
```

IIUC the "else" case is just for boolean

What do you think about the following change?

Reply:
Agree, it's easier to read.  [Invert empty ternary check](https://github.com/StanfordSpezi/spezi-web-design-system/pull/55/commits/d9db1364c2b1146db31217afc41df9a345b3a43e). I used `isBoolean` from `es-toolkit` though - `typeof` is ugly and not sufficient for complex cases anyway.